daemon/src/projection.rs

```diff
@@ -208,18 +220,76 @@ pub struct Cfd {
 /// - It serves as an aggregate that is hydrated from events.
 ///
 /// This dual-role motivates the existence of this struct.
-#[derive(Default, Clone, Debug)]
+#[derive(Clone, Debug)]
 struct Aggregated {
     /// If this is present, we have an active DLC.
     latest_dlc: Option<Dlc>,
     /// If this is present, it should have been published.
     collab_settlement_tx: Option<(Transaction, Script)>,
     /// If this is present, it should have been published.
     cet: Option<Transaction>,
-    closing_price: Option<Price>,
+
+    /// If this is present the cet has not been published
+    timelocked_cet: Option<Transaction>,
 
     commit_published: bool,
     refund_published: bool,
+
+    role: Role,
+}
+
+impl Aggregated {
+    fn new(role: Role) -> Self {
+        Self {
+            latest_dlc: None,
+            collab_settlement_tx: None,
+            cet: None,
+            timelocked_cet: None,
+            commit_published: false,
+            refund_published: false,
+            role,
+        }
+    }
+
+    fn payout(self) -> Option<Amount> {
+        if let Some((tx, script)) = self.collab_settlement_tx {
+            return Some(extract_payout_amount(tx, script));
+        }
+
+        match (self.cet, self.timelocked_cet) {
```

Comment:
Can you explain this logic please: 

From the code comments I understand that : 

- If `self.cet` is present, the `cet` should have been published.
- If `self.timelocked_cet` is present, the `cet` has not been published

What if we don't know which `cet` will be published, both are `None` right? 

If this is the case, than this function returns `None` which leads to `self.profit_btc = profit_btc;` not being set (line 501+).  
Am I correct and there is a problem here?

Reply:
The current logic is, that we set the profit based on the current price when calling `projection::Cfd::new` and then *overwrite* that profit in case we have already know the payout based on a transaction spending from `lock`. 

Returning `None` means that we don't have a fixed payout yet, but keep the initially set payout based on the current price.

Reply:
I missed that. Got it, thanks for the explanation.